AfrikaBurn/MapViewController.swift

```diff
@@ -0,0 +1,339 @@
+    let boundingMapRect: MKMapRect = BurnMap.boundingMapRect
+    var coordinate: CLLocationCoordinate2D = CLLocationCoordinate2DMake(19.751633252193361, -32.326217923051772)
+    
+    func canReplaceMapContent() -> Bool {
+        return true
+    }
+}
+
+class BurnMapOverlayView: MKOverlayRenderer {
+    let overlayImage: UIImage
+    
+    init(overlay: MKOverlay, overlayImage: UIImage) {
+        self.overlayImage = overlayImage
+        super.init(overlay: overlay)
+    }
+    
+    override func draw(_ mapRect: MKMapRect, zoomScale: MKZoomScale, in context: CGContext) {
+        guard let imageRef = overlayImage.cgImage else {
+            return
+        }
+        let theMapRect = overlay.boundingMapRect
+        let theRect = rect(for: theMapRect)
+        
+        context.scaleBy(x: 1, y: -1)
+        context.translateBy(x: 0, y: -theRect.height)
+        context.draw(imageRef, in: theRect)
+    }
+}
+
+
+extension MapViewController: MKMapViewDelegate {
+    
+    
+    func mapView(_ mapView: MKMapView, viewFor annotation: MKAnnotation) -> MKAnnotationView? {
+        
+        guard let abAnnotation = annotation as? ABAnnotation else {
+            return nil
+        }
+        
+        let annotationView: MKAnnotationView
+        if let reusedAnnotationView = mapView.dequeueReusableAnnotationView(withIdentifier: "Pin") {
+            annotationView = reusedAnnotationView
+            annotationView.annotation = annotation
+        } else {
+            annotationView = MKAnnotationView(annotation: annotation, reuseIdentifier: "Pin")
+            annotationView.canShowCallout = true
+        }
+        
+        
+        
+        // if it's a burn annotation add a burn callout
+        if (abAnnotation is BurnAnnotation){
+            let burnAnnotation = abAnnotation as! BurnAnnotation
+            let blurb = UILabel(frame: CGRect(x: 0,y: 0,width: annotationView.frame.size.width,height: 200))
+            blurb.font = UIFont.preferredFont(forTextStyle: .body)
+            blurb.text = burnAnnotation.element.shortBlurb
+            blurb.numberOfLines = 6
+            annotationView.detailCalloutAccessoryView = blurb
+            
+            // Left Accessory
+            // We could make this change the icon based on the element type. Eg. mutant vehicles etc.
+//            annotationView.leftCalloutAccessoryView = UIImageView(image: burnAnnotation.element.elementType.iconImage)
+            
+            
+            let button = UIButton(type: .detailDisclosure)
+            annotationView.rightCalloutAccessoryView = button
+            
+        } else if (abAnnotation is CustomLocationAnnotation) {
+//            let customAnnotation = abAnnotation as! CustomLocationAnnotation
+//            let image = customAnnotation.customLocation.isHomeCamp ? #imageLiteral(resourceName: "home") : #imageLiteral(resourceName: "star")
+//            annotationView.leftCalloutAccessoryView = UIImageView(image: image)
```

Comment:
I could imagine us using these icons in future so might leave it, especially if we use custom callouts that fire from the bottom, the icon might look better in there.